lightning/src/offers/invoice_request.rs

```diff
@@ -1284,6 +1286,30 @@ impl TryFrom<Vec<u8>> for UnsignedInvoiceRequest {
 	}
 }
 
+impl AsRef<[u8]> for InvoiceRequest {
+	fn as_ref(&self) -> &[u8] {
+		&self.bytes
+	}
+}
+
+impl Bech32Encode for InvoiceRequest {
+	const BECH32_HRP: &'static str = "lnr";
```

Comment:
Similarly here. Additionally, this clashes with `Refund`, which is essentially an `invoice_request` without a preceding `offer`.